File: Leetcode_495.java
public class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        if (timeSeries.length == 0) return 0;

        int total = 0;

        for (int i = 0; i < timeSeries.length - 1; i++) {
            int diff = timeSeries[i + 1] - timeSeries[i];
            total += Math.min(duration, diff);
        }

        // Add duration for the last attack
        total += duration;

        return total;
    }
}

File: Leetcode_495.py
from typing import List

class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        if not timeSeries:
            return 0

        total = 0

        for i in range(len(timeSeries) - 1):
            diff = timeSeries[i + 1] - timeSeries[i]
            total += min(duration, diff)

        total += duration  # Add full duration for the last attack
        return total
